[Security] Prevent unneeded implementation of `PasswordHasherAwareInterface` when migrating passwords (#17641)
symfony/symfony-docs: security/passwords.rst

        }
    }

> .. named-password-hashers:

Reviewer: ```suggestion
.. _named-password-hashers:
```

Reviewer 2: Thanks

        }
    }

> Dynamic Password Hashers

Reviewer 2: ```suggestion
.. named-password-hashers:

Dynamic Password Hashers
```

To keep the previous anchor working

        }
    }

.. note::

    When migrating passwords, you don't need to implement ``PasswordHasherAwareInterface``
    to return the legacy hasher name:

>     Symfony will detect it from your ``migrate_from`` configuration.

Reviewer: Let's be even more direct :)

```suggestion
.. caution::

    When :ref:`migrating passwords <security-password-migration>`, you should not
    implement ``PasswordHasherAwareInterface``. Symfony will detect the legacy
    hasher automatically from your ``migrate_from`` configuration.
```

Author: I thought that initially, but it can be legit to implement `PasswordHasherAwareInterface` while migrating passwords; that’s why I specifically warned about returning the legacy hasher name. That being said this should rarely happen :thinking:

Reviewer: Yeah, and if I read https://github.com/symfony/symfony/issues/48348#issuecomment-1328810098 correctly using both features together can create some confusing situations as it might by-pass the migrating hasher.

Author: Yes in fact you must not implement `PasswordHasherAwareInterface` **to** migrate passwords, but you can implement it **while** migrating passwords. That is, the two are de-correlated but you can shoot you in the foot thinking they’re not.

Do you still think we should always discourage implementing `PasswordHasherAwareInterface`?
